### PR TITLE
add keyboard: handwired/oem_ansi_fullsize

### PR DESCRIPTION
**Pull request by johnstcn**
## Description

Added a keyboard+layout for a hand-wired 100% full-sized ANSI keyboard using a Teensy++ 2.0.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/oem_ansi_fullsize/readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# oem_ansi_fullsize
+
+![oem_ansi_fullsize](https://imgur.com/a/Tw7dwK4)
+
+*Description*
+
```
Comment (zvecr) [inline keyboards/handwired/oem_ansi_fullsize/readme.md:6]:
```suggestion

```

File: keyboards/handwired/oem_ansi_fullsize/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = at90usb1286
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
```
Comment (zvecr) [inline keyboards/handwired/oem_ansi_fullsize/rules.mk:5]:
```suggestion
BOOTLOADER = halfkay
```

File: keyboards/handwired/oem_ansi_fullsize/info.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "keyboard_name": "oem_ansi_fullsize",
+    "url": "",
+    "maintainer": "Cian Johnston",
+    "width": 3,
+    "height": 2,
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/handwired/oem_ansi_fullsize/info.json:8]: This doesnt match the layout macro so will be spitting out api/`qmk info` issues

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oem_ansi_fullsize/info.json (hunk 3 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/info.json:8]: Is there a nice way to generate this from the existing source code, or would I be better off just adapting the `info.json` of `handwired/oem_iso_fullsize`?

Comment (zvecr) [reply keyboards/handwired/oem_ansi_fullsize/info.json]: > EDIT: yep, qmk c2json

Nope, thats for a different purpose. https://qmk.fm/converter/ or use `layouts/default/fullsize_iso/info.json` as a base.

File: keyboards/handwired/oem_ansi_fullsize/info.json [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  "keyboard_name": "oem_ansi_fullsize",
+  "url": "https://github.com/johnstcn/qmk",
+  "maintainer": "johnstcn",
+  "width": 22,
+  "height": 6,
+  "layouts": {
+    "LAYOUT": {
+      "layout": [
+        [
+          "KC_ESC",
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/info.json:11]: This file is incorrect. you need key positions, not keycodes, here.

File: keyboards/handwired/oem_ansi_fullsize/config.h [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+ *    R2 <== F3 │o  A2 o  o A6  o│ C3 ==> C0
+ *    R1 <== F4 │o  A3 o  o A7  o│ C4
+ *    R0 <== F5 │o              o│ C5
+ *           F6 │o              o│ C6
+ *           F7 │o  o   o   o   o│ C7
+ *              └────────────────┘
+ *                  R   G   +
+ *                  S   N   5
+ *                  T   D   V
+ */
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/config.h:71]: This probably belongs more in the readme, tbh.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oem_ansi_fullsize/info.json (hunk 4 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/info.json]: Done

File: keyboards/handwired/oem_ansi_fullsize/config.h (hunk 5 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/config.h:71]: Done

File: keyboards/handwired/oem_ansi_fullsize/info.json (hunk 3 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/info.json:8]: Done

File: keyboards/handwired/oem_ansi_fullsize/oem_ansi_fullsize.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ *     R4  │   K41   ││K42││K43││K44││K45││K46││K47││K48││K49││K4A││K4B││    K4D    │       │K4G│       │K4I││K4J││K4K││   │
+ *         └─────────┘└───┘└───┘└───┘└───┘└───┘└───┘└───┘└───┘└───┘└───┘└───────────┘       └───┘       └───┘└───┘└───┘│K4L│
+ *         ┌────┐┌────┐┌────┐┌──────────────────────────────┐┌────┐┌────┐┌────┐┌────┐  ┌───┐┌───┐┌───┐  ┌────────┐┌───┐│   │
+ *     R5  │K50 ││K51 ││K53 ││               K56            ││K5A ││K5B ││K5D ││K5E │  │K5F││K5G││K5H│  │   K5I  ││K5K││   │
+ *         └────┘└────┘└────┘└──────────────────────────────┘└────┘└────┘└────┘└────┘  └───┘└───┘└───┘  └────────┘└───┘└───┘
+ *
+ */
+
+// clang-format off
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/oem_ansi_fullsize.h:57]:
```suggestion
#define LAYOUT_fullsize_ansi( \
```

File: keyboards/handwired/oem_ansi_fullsize/rules.mk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = fullsize_ansi
```

File: keyboards/handwired/oem_ansi_fullsize/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_fullsize_ansi(
```

File: keyboards/handwired/oem_ansi_fullsize/info.json [hunk 9]
```diff
@@ -0,0 +1,122 @@
+{
+    "keyboard_name": "oem_ansi_fullsize",
+    "url": "https://github.com/johnstcn/qmk",
+    "maintainer": "johnstcn",
+    "width": 22.5,
+    "height": 6.25,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/info.json:8]:
```suggestion
        "LAYOUT_fullsize_ansi": {
```

File: keyboards/handwired/oem_ansi_fullsize/config.h [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ *
```
Comment (fauxpark) [inline keyboards/handwired/oem_ansi_fullsize/config.h:42]:
```suggestion
 */
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/oem_ansi_fullsize/config.h (hunk 10 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/config.h]: 🤦‍♂️

File: keyboards/handwired/oem_ansi_fullsize/oem_ansi_fullsize.h (hunk 6 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/oem_ansi_fullsize.h]: Done, but with `LAYOUT_ansi_fullsize` for consistency with the name of the board.

File: keyboards/handwired/oem_ansi_fullsize/rules.mk (hunk 7 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/rules.mk:22]: `ansi_fullsize`

File: keyboards/handwired/oem_ansi_fullsize/keymaps/default/keymap.c (hunk 8 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/keymaps/default/keymap.c]: `LAYOUT_ansi_fullsize`

File: keyboards/handwired/oem_ansi_fullsize/info.json (hunk 9 above)
Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/info.json]: `LAYOUT_ansi_fullsize`

File: keyboards/handwired/oem_ansi_fullsize/rules.mk (hunk 7 above)
Comment (fauxpark) [reply keyboards/handwired/oem_ansi_fullsize/rules.mk:22]: No.
https://github.com/qmk/qmk_firmware/tree/master/layouts/default/fullsize_ansi

Comment (johnstcn) [reply keyboards/handwired/oem_ansi_fullsize/rules.mk:22]: Done

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**